src/Illuminate/Contracts/Auth/UserProvider.php

```diff
@@ -10,7 +10,7 @@ interface UserProvider
      * @param  mixed  $identifier
      * @return \Illuminate\Contracts\Auth\Authenticatable|null
      */
-    public function retrieveById($identifier);
+    public function retrieveById(mixed $identifier);
```

Comment:
Wouldn't this more accurately be typed as `string|int`?

I think if we're doing this, now's a good time to get slightly more explicit about what types we're accepting, where applicable.

There are probably a number of cases where `mixed` is accurate because we're accepting everything under the sun, we should take a closer look at some of these.

Reply:
We will consider it. Thanks!